src/Controllers/Frontend.php

```diff
@@ -437,14 +385,31 @@ public static function template(Silex\Application $app, $template)
             $template .= '.twig';
         }
 
-        $themePath    = realpath($app['paths']['themepath'] . '/');
-        $templatePath = realpath($app['paths']['themepath'] . '/' . $template);
+        return self::render($app, $template, $template);
+    }
 
-        // Verify that the resulting template path is located in the theme directory
-        if ($themePath !== substr($templatePath, 0, strlen($themePath))) {
-            throw new \Exception("Invalid template: $template");
+    /**
+     * Render a template while wrapping Twig_Error_Loader in 404
+     * in case the template is not found by Twig.
+     *
+     * @param  Silex\Application $app
+     * @param  string            $template   Ex: 'listing.twig'
+     * @param  string            $title      '%s' in "No template for '%s' defined."
+     * @return mixed                         Rendered template
+     */
+    private static function render(Silex\Application $app, $template, $title)
+    {
+        try {
+            return $app['twig']->render($template);
+        } catch (\Twig_Error_Loader $e) {
```

Comment:
`Twig_Error_Syntax` is not handled, it could be added in the future, but it would have to handle the case where the 404 page itself triggers an error (like in `_header.twig`)